src/SmartComponents/CreateImageWizard/CreateImageWizard.js

```diff
@@ -240,34 +252,40 @@ class CreateImageWizard extends Component {
 
     onSave () {
         let requests = [];
-        Object.keys(this.state.uploadDestinations).forEach(provider => {
-            switch (provider) {
-                case 'aws': {
-                    let request = {
-                        distribution: this.state.release,
-                        image_requests: [
-                            {
-                                architecture: this.state.arch,
-                                image_type: 'ami',
-                                upload_requests: [ this.state.uploadAWS ],
-                            }],
-                        customizations: {
-                            subscription: this.state.subscription,
-                            packages: this.state.packagesSelectedNames,
-                        },
-                    };
-                    requests.push(request);
-                    break;
-                }
+        if (this.state.uploadDestinations.aws) {
+            let request = {
+                distribution: this.state.release,
+                image_requests: [
+                    {
+                        architecture: this.state.arch,
+                        image_type: 'ami',
+                        upload_requests: [ this.state.uploadAWS ],
+                    }],
+                customizations: {
+                    subscription: this.state.subscription,
+                    packages: this.state.packagesSelectedNames,
+                },
+            };
+            requests.push(request);
+        }
 
-                case 'azure':
-                    break;
-                case 'google':
-                    break;
-                default:
-                    break;
-            }
-        });
+        if (this.state.uploadDestinations.google) {
+            const upload_google = this.state.uploadGoogle;
+            delete upload_google.accountType;
+            let request = {
+                distribution: this.state.release,
+                image_requests: [
+                    {
+                        architecture: this.state.arch,
+                        image_type: 'gcp',
+                        upload_requests: [ upload_google ],
```

Comment:
Right this is getting a little confusing. I think it's best if we don't consider the upload (options) in the state to be directly usable in the request. And build the upload_request here in the onSave:

```
...
upload_requests: [{
    type: 'gcp',
    options: {
        share_with_accounts: upload_google.options.share_with_accounts
    }
}]
...
```

Can do that in a followup though because AWS would need the same treatment. That way we could also just flatten the upload state to just have `upload_google.share_with_accounts`, and not mess with the options all the time.

Reply:
Tracked in #100